src/widget-core/middleware/core.ts

```diff
@@ -0,0 +1,11 @@
+import { create } from '../tsx';
+import { getInvalidator } from '../vdom';
+
+const createFactory = create();
+
+export const invalidator = createFactory(({ id }) => {
```

Comment:
starting to wonder whether we should put these base ones in vdom itself? then we don't have to export the function itself?

Reply:
I think this is a good idea, will have more control in vdom itself also.

Reply:
Done